### PR TITLE
ref(hackweek): inline all list casts for compat map, filter, zip

### PR DESCRIPTION
**Pull request by joshuarli**
This is the 1st step in https://github.com/getsentry/sentry/pull/28058: the brute force inline pass. Saves a function call across the board.

In future steps, I'm going to do my optimization passes to identify and remove as many unnecessary list casts.

- https://github.com/getsentry/sentry/pull/28081
- https://github.com/getsentry/sentry/pull/28082
- https://github.com/getsentry/sentry/pull/28084

I should have done this right after py3, but I was worried about things drifting and didn't have any custom powertooling at the ready at the time. I do now.

### CODE REVIEW

**Review by evanh — APPROVED** [review]
LGTM

File: src/sentry/api/bases/organization.py [hunk 1]
```diff
@@ -166,7 +165,7 @@ def get_requested_project_ids(self, request):
         permission checking, use ``get_projects``, instead.
         """
         try:
-            return set(map(int, request.GET.getlist("project")))
+            return set(list(map(int, request.GET.getlist("project"))))
```
Comment (untitaker) [inline src/sentry/api/bases/organization.py:168]: Can we come up with a mass-transform that avoids intermediate list if we know it will end up in another collection ctor immediately?

File: src/sentry/api/endpoints/assistant.py [hunk 2]
```diff
@@ -10,15 +10,14 @@
...
-from sentry.utils.compat import zip
 
 VALID_STATUSES = frozenset(("viewed", "dismissed"))
 
 
 class AssistantSerializer(serializers.Serializer):
     guide = serializers.CharField(required=False)
     guide_id = serializers.IntegerField(required=False)
-    status = serializers.ChoiceField(choices=zip(VALID_STATUSES, VALID_STATUSES))
+    status = serializers.ChoiceField(choices=list(zip(VALID_STATUSES, VALID_STATUSES)))
```
Comment (untitaker) [inline src/sentry/api/endpoints/assistant.py:20]: VALID_STATUSES is a set so k->v mapping is unstable, right? or does this new ordered-ness of dicts also apply to sets?

File: src/sentry/api/bases/organization.py (hunk 1 above)
Comment (untitaker) [inline src/sentry/api/bases/organization.py:168]: Can we come up with a mass-transform that avoids intermediate list if we know it will end up in another collection ctor immediately?

File: src/sentry/api/endpoints/organization_stats.py [hunk 3]
```diff
@@ -51,7 +50,7 @@ def get(self, request, organization):
             raise ValueError("Invalid group: %s" % group)
 
         if "id" in request.GET:
-            id_filter_set = frozenset(map(int, request.GET.getlist("id")))
+            id_filter_set = frozenset(list(map(int, request.GET.getlist("id"))))
```
Comment (untitaker) [inline src/sentry/api/endpoints/organization_stats.py:53]: Same here, can we do a mass transform here similar to set(list(...))

File: src/sentry/api/event_search.py [hunk 4]
```diff
@@ -353,7 +352,7 @@ class SearchFilter(NamedTuple):
     value: SearchValue
 
     def __str__(self):
-        return "".join(map(str, (self.key.name, self.operator, self.value.raw_value)))
+        return "".join(list(map(str, (self.key.name, self.operator, self.value.raw_value))))
```
Comment (untitaker) [inline src/sentry/api/event_search.py:355]: same here, `join(list(..))` can be `join(..)` afaik

File: src/sentry/api/endpoints/project_stacktrace_link.py [hunk 5]
```diff
@@ -51,8 +50,8 @@ def get(self, request, project):
         # no longer feature gated and is added as an IntegrationFeature
         result["integrations"] = [
             serialize(i, request.user)
-            for i in filter(
-                lambda i: i.has_feature(IntegrationFeatures.STACKTRACE_LINK), integrations
+            for i in list(
```
Comment (untitaker) [inline src/sentry/api/endpoints/project_stacktrace_link.py:53]: Note to self: I think this cannot be simplified further because we may end up with "collection modified during iteration"

File: src/sentry/snuba/discover.py [hunk 6]
```diff
@@ -1116,7 +1115,7 @@ def find_histogram_min_max(fields, min_value, max_value, user_query, params, dat
 
     if min_value is None:
         min_values = [row[get_function_alias(column)] for column in min_columns]
-        min_values = list(filter(lambda v: v is not None, min_values))
+        min_values = list(list(filter(lambda v: v is not None, min_values)))
```
Comment (untitaker) [inline src/sentry/snuba/discover.py:1118]: This seems really unnecessary.

**Review by untitaker — APPROVED** [review]
seems ok but there's some results in there that we probably could handle with another mass-transform

File: src/sentry/api/endpoints/group_similar_issues.py [hunk 7]
```diff
@@ -54,7 +53,7 @@ def get(self, request, group):
 
         # We need to preserve the ordering of the Redis results, as that
         # ordering is directly shown in the UI
-        for group_id, scores in zip(group_ids, group_scores):
+        for group_id, scores in list(zip(group_ids, group_scores)):
```
Comment (wedamija) [inline src/sentry/api/endpoints/group_similar_issues.py:56]: Why is this necessary? Isn't it preferable to just use the iterator here rather than allocate another list?

File: src/sentry/api/endpoints/organization_code_mappings.py [hunk 8]
```diff
@@ -154,7 +153,7 @@ def get(self, request, organization):
 
         # front end handles ordering
         # TODO: Add pagination
-        data = map(lambda x: serialize(x, request.user), queryset)
+        data = list(map(lambda x: serialize(x, request.user), queryset))
         return self.respond(data)
```
Comment (wedamija) [inline src/sentry/api/endpoints/organization_code_mappings.py:157]: Also don't understand why this is an improvement?

**Review by wedamija — COMMENTED** [review]
I'm not totally sure what we gain for a lot of this, could you explain the reasoning? It seems like in a lot of cases we're converting an iterator into a list, but I'm not sure I understand the value of allocating a whole list to then iterate over?

**Comment by joshuarli** [comment]
> I'm not totally sure what we gain for a lot of this, could you explain the reasoning? It seems like in a lot of cases we're converting an iterator into a list, but I'm not sure I understand the value of allocating a whole list to then iterate over?

We're already doing that with the compat. This is just inlining the compat calls in the dumbest, safest way which saves a function call across the board. The idea was to get this in first and have the optimization passes ready in a separate PR, but it looks like we prefer one big PR.

File: src/sentry/api/endpoints/organization_stats.py (hunk 3 above)
Comment (joshuarli) [reply src/sentry/api/endpoints/organization_stats.py:53]: This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834

File: src/sentry/snuba/discover.py (hunk 6 above)
Comment (joshuarli) [reply src/sentry/snuba/discover.py:1118]: This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834

File: src/sentry/api/event_search.py (hunk 4 above)
Comment (joshuarli) [reply src/sentry/api/event_search.py:355]: This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834

File: src/sentry/api/bases/organization.py (hunk 1 above)
Comment (joshuarli) [reply src/sentry/api/bases/organization.py:168]: This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834

File: src/sentry/api/endpoints/group_similar_issues.py (hunk 7 above)
Comment (joshuarli) [reply src/sentry/api/endpoints/group_similar_issues.py:56]: This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834

File: src/sentry/api/endpoints/organization_code_mappings.py (hunk 8 above)
Comment (joshuarli) [reply src/sentry/api/endpoints/organization_code_mappings.py:157]: This is done in an optimization pass that I linked in the description. I guess we'd prefer it to be all rolled up in 1 PR though. https://github.com/getsentry/sentry/pull/28080#issuecomment-908578834
